### DISCUSSION TITLE
Can’t run spoof-mac without errors

### DISCUSSION THREAD

**Discussion by DominikHoffmann** [Homebrew/discussions#5551] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.14
ORIGIN: https://github.com/Homebrew/brew
HEAD: 39d51a45e89ddb4572494c845522509e3deeeb61
Last commit: 4 days ago
Core tap JSON: 11 Aug 20:10 UTC
Core cask tap JSON: 11 Aug 20:10 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
  Homebrew/homebrew-cask-drivers
Untap them with `brew untap`.
```

### Description of issue

I tried to run the command

`sudo spoof-mac set xx:xx:xx:xx:xx:xx en0`

where _en0_ is my Wi-Fi interface. xx:xx:xx:xx:xx:xx is the MAC address of the Wi-Fi interface of an Xbox which we are trying to get onto a Wi-Fi network requiring authentication through a captive portal. Somehow, the Xbox cannot handle that captive portal, which is why the idea was to try to get the Mac to spoof the MAC address of the Xbox, pose as it, get authenticated, and then go back to its real MAC address through a reboot. Hopefully, that would subsequently allow the Xbox to be accepted onto the network.

However, the output of the above command is

`WARNING: The airport command line tool is deprecated and will be removed in a future release.
For diagnosing Wi-Fi related issues, use the Wireless Diagnostics app or wdutil command line tool.
ifconfig: ioctl (SIOCAIFADDR): Can't assign requested address
Traceback (most recent call last):
  File "/opt/homebrew/bin/spoof-mac", line 138, in <module>
    sys.exit(main(arguments, root_or_admin))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/bin/spoof-mac", line 118, in main
    set_interface_mac(device, target_mac, port)
  File "/opt/homebrew/Cellar/spoof-mac/2.1.1_5/libexec/lib/python3.12/site-packages/spoofmac/interface.py", line 540, in set_interface_mac
    spoofer.set_interface_mac(device, mac, port)
  File "/opt/homebrew/Cellar/spoof-mac/2.1.1_5/libexec/lib/python3.12/site-packages/spoofmac/interface.py", line 450, in set_interface_mac
    subprocess.check_call([
  File "/opt/homebrew/Cellar/python@3.12/3.12.4/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ifconfig', 'en0', 'ether', 'xx:xx:xx:xx:xx:xx']' returned non-zero exit status 1.`

Is this something that can be fixed?

### Relevant casks

spoof-mac

**Comment by gromgit** [comment]
> Is this something that can be fixed?

You'll have to ask the `spoof-mac` developers. The error message you showed looks like a problem with `spoof-mac`, not Homebrew.
